### PR TITLE
Add pytest benchmark

### PR DESCRIPTION
**Pull request by dholth**
### Description

Results so far:

- new checkformat-hex-string is much faster than older version
- checking signatures (generating a new PublicKey each time) is half as fast as re-using the PublicKey each time

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [x] Add / update outdated documentation?

### CODE REVIEW

File: conda_content_trust/cli.py [hunk 1]
```diff
@@ -328,7 +329,6 @@ def interactive_modify_metadata(metadata):
             "and prettier printing of JSON, you may install pygments."
         )
         pygments = None
-        from pprint import pprint
```
Comment (dholth) [inline conda_content_trust/cli.py:331]: linter change

**Comment by beeankha** [comment]
pre-commit.ci autofix

File: tests/test_benchmark.py [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+SAMPLE_HEX_KEY_2 = "deadbeef" * 8
+
+
+def bench_checkformat_hex_string():
+    hex_string = "0123456789abcdef" * 4
+    checkformat_hex_string(hex_string)
+
+
+def test_benchmark_checkformat_hex_string(benchmark):
+    benchmark(bench_checkformat_hex_string)
```
Comment (kenodegard) [inline tests/test_benchmark.py:46]: `benchmark` works like Python's `functools.partial` and accepts args and kwargs:

```suggestion
def test_benchmark_checkformat_hex_string(benchmark):
    hex_string = "0123456789abcdef" * 4
    benchmark(checkformat_hex_string, hex_string)
```

**Comment by dholth** [comment]
Let's merge this.

One way to approach this project would be to continue adding benchmarks until we run out of functions in conda-content-trust, but we should be able to merge any individual PR before reaching that level.

**Review by dbast — APPROVED** [review]

**Pull request merged**